Thingamajig.ts

```diff
@@ -0,0 +1,86 @@
+type Dimensions = {
+  width: number;
+  depth: number;
+  height: number;
+};
+interface IThingamajigHandler {
+  numberHandler: (t: Thingamajig) => void;
+  stringHandler: (t: Thingamajig) => void;
+  objectHandler: (t: Thingamajig) => void;
+}
+enum ThingamajigColor {
+  Red = "RED",
+  Blue = "BLUE",
+  White = "WHITE"
+}
+export default class Thingamajig {
+  name: string;
+  protected dimensions: Dimensions;
+  private label: any;
+  // at a given timestamp calls the function with the Thingamajig
+  private giveMeAtTimestamp: number;
+  private giveMeAt: ((t: Thingamajig) => void) | null;
+  color: ThingamajigColor;
+  constructor(
+    dimension: Dimensions,
+    name: string,
+    label: number | string | object,
+    giveMeAtTimestamp: number,
+    giveMeAt: (t: Thingamajig) => void,
+    c: ThingamajigColor
+  ) {
+    this.dimensions = dimension;
+    this.name = name;
+    this.label = label;
+    this.giveMeAtTimestamp = giveMeAtTimestamp;
+    this.giveMeAt = giveMeAt;
+    const cancel = setInterval(() => {
+      if (new Date().valueOf() > this.giveMeAtTimestamp && this.giveMeAt) {
+        this.giveMeAt(this);
+        this.giveMeAt = null;
+        clearInterval(cancel);
+      }
+    }, 1000);
+    this.color = c;
+  }
+  // a Thingamajig is valid if any of its dimensions is less than 100
+  validDimensions(): boolean {
+    if (this.dimensions.width < 100) {
+      return true;
+    } else if (this.dimensions.depth < 100) {
+      return true;
+    } else if (this.dimensions.height < 100) {
+      return true;
+    }
+    return false;
+  }
```

Comment:
M: I believe this method could be improved by taking the dimensions to validate as an argument, so we can do the validation before update the Thingamajig instance

```suggestion
  // a Thingamajig is valid if any of its dimensions is less than 100
  validDimensions(dimensions: Dimensions): boolean {
    if (dimensions.width < 100) {
      return true;
    } else if (dimensions.depth < 100) {
      return true;
    } else if (dimensions.height < 100) {
      return true;
    }
    return false;
  }
```

C: Also, to make more readable, you could do something like this
```suggestion
  // a Thingamajig is valid if any of its dimensions is less than 100
  validDimensions(dimensions: Dimensions): boolean {
    return Object.values(dimensions).some((dimension) => dimension < 100);
  }
```